src/services/sponsorships.ts

```diff
@@ -19,49 +15,82 @@ const getSponsorshipChainId = () => {
-    initialMinimumStakeWei: number
-    initialMinHorizonSeconds: number
-    initialMinOperatorCount: number
+    minOperatorCount: number
+    maxOperatorCount?: number
+    payoutRate: BNish // tokens per second
+    minimumStakeTime: BNish // seconds
+    initialFunding: BNish // wei
     streamId: string
-    metadata: object
-    policies: Address[]
-    initParams: number[]
+    metadata?: object
 }
 
 export async function createSponsorship({
-    initialMinimumStakeWei,
-    initialMinHorizonSeconds,
-    initialMinOperatorCount,
+    minOperatorCount,
+    maxOperatorCount,
+    payoutRate,
+    minimumStakeTime,
+    initialFunding,
     streamId,
-    metadata,
-    policies,
-    initParams,
-}: SponsorshipParams) {
+    metadata = {},
+}: SponsorshipParams): Promise<void> {
     const chainId = getSponsorshipChainId()
 
     const chainConfig = getConfigForChain(chainId)
 
+    const paymentTokenSymbolFromConfig = getCoreConfig().sponsorshipPaymentToken as string
+
     await networkPreflight(chainId)
 
     const signer = await getSigner()
 
-    const factory = new Contract(
-        chainConfig.contracts['SponsorshipFactory'],
-        sponsorshipFactoryABI,
+    const policies = [
+        chainConfig.contracts.SponsorshipStakeWeightedAllocationPolicy,
+        chainConfig.contracts.SponsorshipDefaultLeavePolicy,
+    ]
+
+    const policyParams = [payoutRate.toString(), minimumStakeTime.toString()]
+
+    if (maxOperatorCount) {
+        policies.push(chainConfig.contracts.SponsorshipMaxOperatorsJoinPolicy)
+        policyParams.push(maxOperatorCount.toString())
+    }
+
+    const data = defaultAbiCoder.encode(
+        ['uint', 'uint32', 'uint32', 'string', 'string', 'address[]', 'uint[]'],
+        [
+            parseEther('60'), // initialMinimumStakeWei - hardcoded for now
+            0, // initialMinHorizonSeconds - hardcoded for now
+            minOperatorCount,
+            streamId,
+            JSON.stringify(metadata),
+            policies,
+            policyParams,
+        ],
+    )
+
+    const token = new Contract(
+        chainConfig.contracts[paymentTokenSymbolFromConfig],
+        ERC677ABI,
         signer,
-    ) as SponsorshipFactory
-
-    const tx = await factory.deploySponsorship(
-        initialMinimumStakeWei,
-        initialMinHorizonSeconds,
-        initialMinOperatorCount,
-        streamId,
-        JSON.stringify(metadata),
-        policies,
-        initParams,
+    ) as ERC677
+
+    const sponsorshipDeployTx = await token.transferAndCall(
+        chainConfig.contracts['SponsorshipFactory'],
+        initialFunding.toString(),
+        data,
     )
 
-    await tx.wait()
+    const sponsorshipDeployReceipt = await sponsorshipDeployTx.wait()
+
+    /* You may wonder why we are accessing the SECOND transfer event, that's because first the funds are being transferred to
+     *  the sponsorship factory and then the factory transfers them to the sponsorship contract */
```

Comment:
Don't like the formatting of this.
```suggestion
    /**
     * You may wonder why we are accessing the SECOND transfer event, that's because first
     * the funds are being transferred to the sponsorship factory and then the factory
     * transfers them to the sponsorship contract.
     */
```